From ccb66689c0e04db4ac8474e573ca2e5f055f5d7b Mon Sep 17 00:00:00 2001
From: Daniel Drake <dan@reactivated.net>
Date: Tue, 22 Jun 2010 18:19:15 -0500
Subject: [PATCH] Add log message for open permission denied errors

This message was previously in libusb but is being removed (ticket #44).
Move it here so that this caveat (also documented in README) is more
obvious.
---
 libusb/core.c |    4 ++++
 1 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/libusb/core.c b/libusb/core.c
index 62fae8c..3da15c6 100644
--- a/libusb/core.c
+++ b/libusb/core.c
@@ -659,6 +659,10 @@ API_EXPORTED usb_dev_handle *usb_open(struct usb_device *dev)
 
 	r = libusb_open((libusb_device *) dev->dev, &udev->handle);
 	if (r < 0) {
+		if (r == LIBUSB_ERROR_ACCESS) {
+			usbi_info("Device open failed due to a permission denied error.");
+			usbi_info("libusb requires write access to USB device nodes.");
+		}
 		usbi_err("could not open device, error %d", r);
 		free(udev);
 		errno = libusb_to_errno(r);
-- 
1.7.2.2

